src/utils/algolia-queries.ts

```diff
@@ -135,7 +135,22 @@ const queries = [
           existingProblem.tags = [
             ...new Set([...existingProblem.tags, ...(node.tags || [])]),
           ];
-          if (moduleInfo) existingProblem.problemModules.push(moduleInfo);
+          if (moduleInfo) {
+            // https://stackoverflow.com/questions/2218999/how-to-remove-all-duplicates-from-an-array-of-objects
+            const removeDuplicates = arrayOfObjects => {
+              return arrayOfObjects.filter(
+                (object, index) =>
+                  index ===
+                  arrayOfObjects.findIndex(
+                    obj => JSON.stringify(obj) === JSON.stringify(object)
+                  )
+              );
+            };
+            existingProblem.problemModules = removeDuplicates([
```

Comment:
for some reason, this didn't have any effect locally. is this expected? @thecodingwizard

Reply:
Yes. The process to update algolia is separate (and I forgot how... I think it updates on `gatsby build` if certain environment variables are set. See https://github.com/algolia/gatsby-plugin-algolia and https://github.com/cpinitiative/usaco-guide/blob/master/gatsby-config.ts#L116 . There should be a fast way to test this but I forgot...)

Reply:
I think a more efficient way to do this is to check if `moduleInfo` exists in `existingProblem.problemModules` already; if so, do nothing, if not, add. Can probably be done with something like `existingProblem.problemModules.find(module => module.id === moduleInfo.id)` or something (`find` returns `undefined` if no match is found)